modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageStore.java

```diff
@@ -106,6 +106,25 @@ DurableStoreConnection initializeMessageStore(AndesContextStore contextStore,
     void moveMetadataToDLC(List<AndesMessageMetadata> messages, String dlcQueueName) throws AndesException;
 
     /**
+     * Method to move a message to dead letter channel when the expiry check on DLC is enabled
+     * by configuration
+     * @param messageId    the message id to move
+     * @param dlcQueueName the dead letter channel queue name for the message to be moved
+     * @throws AndesException
+     */
+    void moveMetadataToDLCWhenExpiryCheckEnabled(long messageId, String dlcQueueName) throws AndesException;
```

Comment:
How about :
 this method name refers to a configuration. could you put a name based on the intended behaviour rather then focusing on the configuration value (or what the configuration means)